File: Web/RMA-Web-Shell/projects/clientcare/src/app/policy-manager/shared/entities/category-insured-cover.ts
import { CategoryInsuredEnum } from "../enums/categoryInsuredEnum";

export class CategoryInsuredCover {
  coverId: number;
  policyId: number;
  categoryInsured: CategoryInsuredEnum;
  effectiveFrom: Date;
  effectiveTo: Date;

  isDeleted: boolean;
  createdBy: string;
  createdDate: Date;
  modifiedBy: string;
  modifiedDate: Date;
}

File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/Policy/CategoryInsuredCover.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.ClientCare.Contracts.Entities.Policy
{
    public class CategoryInsuredCover
    {
        public int CategoryInsuredCoverId { get; set; } // CategoryInsuredCoverId (Primary key)
        public int PolicyId { get; set; } // PolicyId
        public CategoryInsuredEnum CategoryInsured { get; set; } // CategoryInsuredId
        public System.DateTime EffectiveFrom { get; set; } // EffectiveFrom
        public System.DateTime? EffectiveTo { get; set; } // EffectiveTo
        public bool IsDeleted { get; set; } // IsDeleted
        public string CreatedBy { get; set; } // CreatedBy (length: 50)
        public System.DateTime CreatedDate { get; set; } // CreatedDate
        public string ModifiedBy { get; set; } // ModifiedBy (length: 50)
        public System.DateTime ModifiedDate { get; set; } // ModifiedDate
    }
}
